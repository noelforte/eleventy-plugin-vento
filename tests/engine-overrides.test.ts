import { EleventyTest } from './_eleventy-test-instance.js';
import { describe, test } from 'vitest';

const testInstance = new EleventyTest('./tests/stubs-engine-overrides/');

await testInstance.rebuild();

const matrix = ['html', 'md'];

test.for(matrix)('Use Vento as `.%s` engine', { concurrent: true }, async (ext, { expect }) => {
	const result = testInstance.getBuildResultForUrl(`/as-${ext}-engine/`);
	expect(result?.content).toMatchSnapshot(`override-${ext}-engine`);
});
