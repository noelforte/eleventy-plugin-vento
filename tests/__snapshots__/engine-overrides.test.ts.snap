// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Use Vento as \`.html\` engine > override-html-engine 1`] = `
"
<p>2 + 2 = 4</p>

<p>This is a cool page</p>
"
`;

exports[`Use Vento as \`.md\` engine > override-md-engine 1`] = `
"<p>2 + 2 = 4</p>
<p>This is a cool page</p>
"
`;
